File: src/test/java/io/spring/deepdive/HtmlTests.java
package io.spring.deepdive;

import org.junit.Test;
import org.junit.runner.RunWith;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.test.context.junit4.SpringRunner;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.boot.test.context.SpringBootTest.*;

@RunWith(SpringRunner.class)
@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
public class HtmlTests {

    @Autowired
    private TestRestTemplate restTemplate;

    @Test
    public void assertContentOnBlogPage() {
        String body = restTemplate.getForObject("/", String.class);
        assertThat(body)
                .contains("Reactor Bismuth is out")
                .contains("September 28th")
                .contains("Sebastien")
                .doesNotContain("brand-new generation");
    }

    @Test
    public void assertContentOnBlogPostPage() {
        String body = restTemplate.getForObject("/article/spring-framework-5-0-goes-ga", String.class);
        assertThat(body)
                .contains("Spring Framework 5.0 goes GA")
                .contains("Dear Spring community")
                .contains("brand-new generation")
                .contains("Juergen")
                .doesNotContain("Sebastien");
    }

}

File: src/test/kotlin/io/spring/deepdive/HtmlTests.kt
package io.spring.deepdive

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.*
import org.springframework.boot.web.server.LocalServerPort
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.web.reactive.function.client.WebClient

import org.springframework.web.reactive.function.client.bodyToMono
import reactor.test.test

@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
class HtmlTests(@LocalServerPort port: Int) {

    // TODO Migrate to WebTestClient when https://youtrack.jetbrains.com/issue/KT-5464 will be fixed
    protected val client = WebClient.create("http://localhost:$port")

    @Test
    fun `Assert content on blog page`() {
        client.get().uri("/").retrieve().bodyToMono<String>()
                .test()
                .consumeNextWith {
                    assertThat(it)
                            .contains("Reactor Bismuth is out")
                            .contains("September 28th")
                            .contains("Sebastien")
                            .doesNotContain("brand-new generation")
                }.verifyComplete()
    }

    @Test
    fun `Assert content on blog post page`() {
        client.get().uri("/article/spring-framework-5-0-goes-ga").retrieve().bodyToMono<String>()
                .test()
                .consumeNextWith {
                    assertThat(it)
                            .contains("Spring Framework 5.0 goes GA")
                            .contains("Dear Spring community")
                            .contains("brand-new generation")
                            .contains("Juergen")
                            .doesNotContain("Sebastien")
                }.verifyComplete()
    }

}
